src/external-modules/Wechaty.Puppet/WechatyPuppet.cs

```diff
@@ -653,34 +653,41 @@ protected async Task MessagePayloadDirty(string messageId)
 
         public async Task<MessagePayload?> MessagePayload(string messageId)
         {
-            if (Logger.IsEnabled(LogLevel.Trace))
+            try
             {
-                Logger.LogTrace($"messagePayload({messageId})");
-            }
+                if (Logger.IsEnabled(LogLevel.Trace))
+                {
+                    Logger.LogTrace($"messagePayload({messageId})");
+                }
 
-            if (string.IsNullOrWhiteSpace(messageId))
-            {
-                throw new ArgumentException("no id");
-            }
+                if (string.IsNullOrWhiteSpace(messageId))
+                {
+                    throw new ArgumentException("no id");
+                }
 
-            //1. Try to get from cache first
-            var cachedPayload = MessagePayloadCache(messageId);
-            if (cachedPayload != null)
-            {
-                return cachedPayload;
-            }
+                //1. Try to get from cache first
+                var cachedPayload = MessagePayloadCache(messageId);
+                if (cachedPayload != null)
+                {
+                    return cachedPayload;
+                }
 
-            //2. Cache not found
-            var rawPayload = await MessageRawPayload(messageId);
-            var payload = MessageRawPayloadParser(rawPayload);
+                //2. Cache not found
+                var rawPayload = await MessageRawPayload(messageId);
+                var payload = MessageRawPayloadParser(rawPayload);
 
-            CacheMessagePayload.Set(messageId, payload);
-            if (Logger.IsEnabled(LogLevel.Trace))
+                CacheMessagePayload.Set(messageId, payload);
+                if (Logger.IsEnabled(LogLevel.Trace))
+                {
+                    Logger.LogTrace($"messagePayload({messageId}) cache SET");
+                }
+
+                return payload;
+            }
+            catch (Exception ex)
             {
-                Logger.LogTrace($"messagePayload({messageId}) cache SET");
+                throw ex;
```

Comment:
don't catch and rethrow exception like ```throw ex```, this will change the stack of exception.